docs/development/code_guide.rst

```diff
@@ -49,6 +49,10 @@ Coding Style
 * Use ``Optional[type]`` for type hinted keyword arguments with a
   default value of ``None``.
 
+* Avoid using `lambda`.  Anonymous functions created with `lambda`
+  cannot be pickled, and `lambda` notation may be unfamiliar to
```

Comment:
That lambdas cannot be pickled is of course true; but as far as I know, and ["Is there an easy way to pickle a python function (or otherwise serialize its code)?" on Stack](https://stackoverflow.com/questions/1253528/is-there-an-easy-way-to-pickle-a-python-function-or-otherwise-serialize-its-cod) seems to confirm, Python functions just can't be pickled at all, lambda or no lambda. ["How to pickle a python function with its dependencies?"](https://stackoverflow.com/questions/10048061/how-to-pickle-a-python-function-with-its-dependencies) seems to show you can do it using `marshal` or `cloudpickle` packages, but that's not exactly the same thing - I don't think scipy multiprocessing will handle that, for example.